server.go

```diff
@@ -1417,23 +1412,6 @@ func (s *Server) HTTPServe() {
 	graceful.Shutdown()
 }
 
-// gRPCServe starts the gRPC server and blocks until an error is encountered,
-// or the server is shutdown.
-//
-// TODO this doesn't handle SIGUSR2 and SIGHUP on it's own, unlike HTTPServe
-// As long as both are running this is actually fine, as Serve will stop
-// the gRPC server when the HTTP one exits.  When running just gRPC however,
-// the signal handling won't work.
-func (s *Server) gRPCServe() {
-	entry := log.WithFields(logrus.Fields{"address": s.grpcListenAddress})
-	entry.Info("Starting gRPC server")
-	if err := s.grpcServer.Serve(s.grpcListenAddress); err != nil {
-		entry.WithError(err).Error("gRPC server was not shut down cleanly")
-	}
-
-	entry.Info("Stopped gRPC server")
-}
-
 // Try to perform a graceful stop of the gRPC server.  If it takes more than
 // 10 seconds, timeout and force-stop.
 func (s *Server) gRPCStop() {
```

Comment:
Is this something that belongs in the proxy source as well?

Reply:
Ah yes, good catch, thanks!